tck/features/expressions/pattern/Pattern1.feature

```diff
@@ -28,6 +28,395 @@
+
+  Scenario: [17] Matching two nodes on a specific type of variable length undirected connection
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A)-[:REL1]->(b:B), (b)-[:REL2]->(a), (a)-[:REL3]->(:C), (a)-[:REL1]->(:D)
+      """
+    When executing query:
+      """
+      MATCH (n), (m) WHERE (n)-[:REL1*]-(m) RETURN n
+      """
+    Then the result should be, in any order:
+      | n    | m    |
+      | (:A) | (:B) |
+      | (:A) | (:D) |
+      | (:D) | (:B) |
+      | (:B) | (:D) |
+    And no side effects
+
+  Scenario: [18] Matching two nodes on a specific type of undirected connection with length 2
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A)-[:REL1]->(b:B), (b)-[:REL2]->(a), (a)-[:REL3]->(:C), (a)-[:REL1]->(:D)
+      """
+    When executing query:
+      """
+      MATCH (n), (m) WHERE (n)-[:REL1*2]-(m) RETURN n
+      """
+    Then the result should be, in any order:
+      | n    | m    |
+      | (:D) | (:B) |
+      | (:B) | (:D) |
+    And no side effects
+
+
+  Scenario: [19] Using a negated existential pattern predicate
+    Given an empty graph
+    And having executed:
+	"""
+	CREATE (a:A)-[:REL1]->(b:B), (b)-[:REL2]->(a), (a)-[:REL3]->(:C), (a)-[:REL1]->(:D)
+	"""
+    When executing query:
+	"""
+	MATCH (n) WHERE NOT (n)-[:REL2]-() RETURN n
+	"""
+    Then the result should be, in any order:
+      | n    |
+      | (:A) |
+    And no side effects
+
+  Scenario: [20] Using two existential pattern predicates in a conjunction
+    Given an empty graph
+    And having executed:
+	"""
+	CREATE (a:A)-[:REL1]->(b:B), (b)-[:REL2]->(a), (a)-[:REL3]->(:C), (a)-[:REL1]->(:D)
+	"""
+    When executing query:
+	"""
+	MATCH (n) WHERE (n)-[:REL1]-() AND (n)-[:REL3]-() RETURN n
+	"""
+    Then the result should be, in any order:
+      | n    |
+      | (:A) |
+    And no side effects
+
+  Scenario: [21] Using two existential pattern predicates in a disjunction
+    Given an empty graph
+    And having executed:
+	"""
+	CREATE (a:A)-[:REL1]->(b:B), (b)-[:REL2]->(a), (a)-[:REL3]->(:C), (a)-[:REL1]->(:D)
+	"""
+    When executing query:
+	"""
+	MATCH (n) WHERE (n)-[:REL1]-() OR (n)-[:REL2]-() RETURN n
+	"""
+    Then the result should be, in any order:
+      | n    |
+      | (:A) |
+      | (:B) |
+    And no side effects
+
+  Scenario: [22] Fail on using pattern in RETURN projection
+    Given any graph
+    When executing query:
+	"""
+	MATCH (n) RETURN (n)-[]->()
+	"""
+    Then a SyntaxError should be raised at compile time: UnexpectedSyntax
+
+  Scenario: [23] Fail on using pattern in WITH projection
+    Given any graph
+    When executing query:
+	"""
+	MATCH (n) WITH (n)-[]->() RETURN n
+	"""
+    Then a SyntaxError should be raised at compile time: UnexpectedSyntax
+
+Scenario: [24] Fail on using pattern in right-hand side of SET
+    Given any graph
+    When executing query:
+	"""
+	MATCH (n) SET x.prop =  head(node(head((a)-[:REL]->()))).x
```

Comment:
Wrong query syntax (apart from what is intended to be wrong).

- Variable names do not align
- function name has a typo

Fix:

```suggestion
	MATCH (n) SET n.prop = head(nodes(head((n)-[:REL]->()))).foo
```

FYI: With this fix, the scenario does not fail anymore on Neo4j. However, that is legacy behavior. Before it failed, but it failed for the wrong reason (not declared variable, unknown function).